### ISSUE TITLE
Translate.py

### ISSUE THREAD

**Issue by ajayjadi** [ajayjadi/FAQ_Chatbot#1]
import torch
from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
from IndicTransToolkit import IndicProcessor
import json

# Check if CUDA is available and set the device accordingly
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

# Initialize the IndicProcessor for inference
ip = IndicProcessor(inference=True)

# Load the translation model and tokenizer
model_name = "ai4bharat/indictrans2-en-indic-dist-200M"
tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
model = AutoModelForSeq2SeqLM.from_pretrained(model_name, trust_remote_code=True)

# Move the model to the GPU if available
model.to(device)

# Translate the question and choices
def translate_text(text, src_lang="eng_Latn", tgt_lang="tel_Telu"):
    # Preprocess the text using the IndicProcessor
    batch = ip.preprocess_batch([text], src_lang=src_lang, tgt_lang=tgt_lang)
    
    # Tokenize the batch and generate input encodings
    batch = tokenizer(batch, padding="longest", truncation=True, max_length=256, return_tensors="pt")
    
    # Move the batch to the GPU if available
    batch = {key: value.to(device) for key, value in batch.items()}
    
    # Use the model to generate the translated text
    with torch.no_grad():
        outputs = model.generate(**batch, num_beams=5, num_return_sequences=1, max_length=256)

    # Decode the generated tokens into text
    with tokenizer.as_target_tokenizer():
        outputs = tokenizer.batch_decode(outputs, skip_special_tokens=True, clean_up_tokenization_spaces=True)
    
    # Postprocess the translations
    translations = ip.postprocess_batch(outputs, lang=tgt_lang)
    return translations[0]

# Load the input data from the arc_challenge_train.json file
with open('arc_challenge_train.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

# Translate each entry
translated_data = []
i=0
for entry in data:
    translated_question = translate_text(entry["question"])
    translated_choices = [translate_text(choice) for choice in entry["choices"]["text"]]

    # Create the translated entry with both text and label intact
    translated_entry = {
        "id": entry["id"],
        "question": translated_question,
        "choices": translated_choices,
        "choices_label": entry["choices"]["label"],  # Keep the original labels
        "answerKey": entry["answerKey"]
    }
    print(translated_entry)
    translated_data.append(translated_entry)
    print(i+1)
    i+=1

# Save the translated data to a jsonl file
with open('1_translated_arc_challenge_train.jsonl', 'w', encoding='utf-8') as f:
    for entry in translated_data:
        json.dump(entry, f, ensure_ascii=False)
        f.write("\n")  # Ensure each entry is on a new line

print(f"Translated data saved to 'translated_arc_challenge_train.jsonl'.")
